core/shared/src/main/scala/zio/metrics/MetricClient.scala

```diff
@@ -43,4 +44,13 @@ private[zio] object MetricClient {
    */
   final def snapshot()(implicit unsafe: Unsafe): Set[MetricPair.Untyped] =
     metricRegistry.snapshot()
+
+  trait Listener {
+    def updateHistogram(key: MetricKey[MetricKeyType.Histogram], value: Double): UIO[Unit]
+    def updateGauge(key: MetricKey[MetricKeyType.Gauge], value: Double): UIO[Unit]
+    def updateFrequency(key: MetricKey[MetricKeyType.Frequency], value: String): UIO[Unit]
+    def updateSummary(key: MetricKey[MetricKeyType.Summary], value: (Double, java.time.Instant)): UIO[Unit]
```

Comment:
Might be better to avoid the tuple `(Double, java.time.Instant)` and instead pass two parameters.